xrayvision/Visibility.py

```diff
@@ -412,3 +426,89 @@ def convert_units_to_tex(string: str):
                 final_string += string[i]
         final_string += opened * "}"
         return final_string
+
+    @staticmethod
+    def from_fits_file(path):
+        """
+        Creates RHESSIVisibility objects from compatible fits files
+
+        Parameters
+        ----------
+        path: str
+            Path where the fits file can be found
+
+        Examples
+        --------
+
+        Notes
+        -----
+        It separates the Visibility data based on the time and energy
+        ranges.
+        """
+        hudlist = fits.open(path)
+        for i in hudlist:
+            if i.name == "VISIBILITY":
+                # Checking how many data structures we have
+                data_sort = {}
+                erange = i.data["erange"]
+                erange_unique = np.unique(erange, axis=0)
+                trange = i.data["trange"]
+                trange_unique = np.unique(trange, axis=0)
+
+                def find_erange(e):
+                    for i, j in enumerate(erange_unique):
+                        if np.allclose(j, e):
+                            return i
+
+                def find_trange(t):
+                    for i, j in enumerate(trange_unique):
+                        if np.allclose(j, t, rtol=1e-15):
+                            return i
+
+                for j, k in enumerate(erange_unique):
+                        data_sort[j] = {}
+
+                for j, k in enumerate(trange):
```

Comment:
I'm a bit worried about performance here iterating over what could be a very large number of values. I think something like `np.where(trange = unique_trange)` could be use iterate over the smaller unique_trange values and then find the matching data. Also I'm pretty sure the data are contiguous so we should be able to use that to our advantage. This is more of a comment for my self late TBH